packages/runtime/client/rest/src/payout/endpoints/collection.ts

```diff
@@ -0,0 +1,73 @@
+
+import { AxiosInstance } from 'axios';
+
+import * as t from "io-ts/lib/index.js";
+import * as TE from 'fp-ts/lib/TaskEither.js'
+import { pipe } from 'fp-ts/lib/function.js';
+import * as E from 'fp-ts/lib/Either.js'
+import * as A from 'fp-ts/lib/Array.js'
+import * as O from 'fp-ts/lib/Option.js';
+import { Newtype, iso } from 'newtype-ts'
+import { formatValidationErrors } from 'jsonbigint-io-ts-reporters'
+import { fromNewtype, optionFromNullable } from 'io-ts-types';
+import { stringify } from 'qs';
+
+import * as HTTP from '@marlowe.io/adapter/http';
+import { DecodingError } from '@marlowe.io/adapter/codec';
+
+import { AssetId, unPolicyId, unContractId } from '@marlowe.io/runtime-core';
+
+import { ContractId } from "@marlowe.io/runtime-core";
+import { PayoutHeader } from '../header.js';
+import { PayoutStatus } from '../status.js';
+
+export interface ContractsRange extends Newtype<{ readonly ContractsRange: unique symbol }, string> {}
+export const ContractsRange = fromNewtype<ContractsRange>(t.string)
+export const unContractsRange =  iso<ContractsRange>().unwrap
+export const contractsRange =  iso<ContractsRange>().wrap
+
+export type GETHeadersByRange = (rangeOption: O.Option<ContractsRange>) 
+    => (contractIds : ContractId[]) 
+    => (roles : AssetId[])
+    => (statusOption : O.Option<PayoutStatus>) 
+    => TE.TaskEither<Error | DecodingError,GETByRangeResponse>
```

Comment:
Not to change now, but if this is something that a final JS user will use I'd recommend using an object instead of curried arguments.

Also, not sure what this function is supposed to do, adding comments would improve its legibility.

Reply:
nope they won't use that